File: 3 Longest Substring without Repeats.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> table;
        int l, r, ans=0;
        for(l=0, r=0; r<s.size(); r++){
            if (table.count(s[r])){
                l = max(table[s[r]], l);
                table[s[r]] = r+1;
            }
            else
                table.insert({s[r], r+1});
            ans = max(ans, r - l + 1);
        }
        return ans;
    }  
};

File: 3 Longest Substring without Repeats.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        table = {}
        ans = 0
        l = 0
        for r, c in enumerate(s):
            if c in table:
                l = max(table[c], l)
                table[c] = r + 1
            else:
                table[c] = r + 1
            ans = max(ans, r - l + 1)
        return ans